XenModel/Utils/Helpers.cs

```diff
@@ -719,12 +719,34 @@ public static DataSourceCategory GetDataSourceCategory(string name)
             return DataSourceCategory.Custom;
         }
 
+        private static string GetDataSourceId(string originalId)
+        {
+            switch (originalId)
+            {
+                case "memory_free_kib":
+                    return "memory_used_kib";

```

Comment:
I think these are the wrong way around

Reply:
The original was performing the same overrides, unless I'm missing something :/ 

`memory_free_kib` 👉 `memory_used_kib`
`memory_internal_free` 👉 `memory_internal_used`

```csharp
if (dataSource.name_label == "memory_free_kib" && xenObject is Host)
                    friendlyName = Helpers.GetFriendlyDataSourceName("memory_used_kib", xenObject);
                else if (dataSource.name_label == "memory_internal_free" && xenObject is VM)
                    friendlyName = Helpers.GetFriendlyDataSourceName("memory_internal_used", xenObject);
                else
                    friendlyName = Helpers.GetFriendlyDataSourceName(dataSource.name_label, xenObject);
```

and

```csharp
if (datasourceName == "memory_free_kib")
                FriendlyName = Helpers.GetFriendlyDataSourceName("memory_used_kib", xo);
            else if (datasourceName == "memory_internal_free")
                FriendlyName = Helpers.GetFriendlyDataSourceName("memory_internal_used", xo);
            else
                FriendlyName = Helpers.GetFriendlyDataSourceName(datasourceName, xo);
```